docs/GUIDE_FOR_KOGITO_DEVS.md

```diff
@@ -0,0 +1,352 @@
+are the specific installation commands/links for Fedora:
+
+### Single Command
+- Maven: `sudo dnf install maven`
+
+### Multiple Commands
+- [GraalVM](https://www.graalvm.org/docs/getting-started/linux/) 
+  - set `JAVA_HOME`/`GRAALVM_HOME` environment variable
+  - [Native Image compilation](https://www.graalvm.org/reference-manual/native-image/#install-native-image) 
+    (after installing GraalVM)
+    ```sh
+    gu install native-image
+    sudo dnf install gcc glibc-devel zlib-devel libstdc++-static
+    ```
+
+# Running Operator Locally
+To test that your changes to the operator work, you will want 
+to be able to run your modified operator locally. To do 
+this, you will need to apply some CRD's/YAML's. To do this, 
+you can run these commands from the base directory of your 
+cloned operator repository after logging into your CRC 
+cluster and choosing a namespace:
+```sh
+./hack/install.sh
+DEBUG=true make run
+```
+
+In future runs, only `DEBUG=true make run` needs to be run. 
+But if no changes were made to the `api/` folder, `DEBUG=true go run main.go` 
+can be run for a faster startup time.
+
+# Building Examples
+To test that your changes to the operator works, you will 
+usually want to deploy a Kogito application to OpenShift. To 
+do so, you will need to first compile [a Kogito example](https://github.com/kiegroup/kogito-examples). 
+
+You can compile Quarkus Kogito applications into either a JAR file 
+or a native executable. The native executable packages all 
+the Java classes and its dependencies into the executable 
+and allows for a quicker startup and lower memory usage in 
+exchange for a longer build time. You can read more about 
+native images in [GraalVM's docs](https://www.graalvm.org/reference-manual/native-image). 
+Spring Boot Kogito applications can only be compiled into a 
+JAR.
+
+Instructions to build specific examples can be found in the 
+README for each example. Generally speaking, JAR's are 
+compiled with `mvn clean package -DskipTests=true` and native executables are 
+compiled with `mvn clean package -Pnative -DskipTests=true` (GraalVM must be 
+installed with native image compilation). To build with 
+additional profile options (e.g. persistence or events), you 
+can separate the profiles with commas: `mvn clean package -Pnative,persistence,events -DskipTests=true`.
+
+# Deploying Examples to CRC
+*Note*: The Kogito operator must be running in the current 
+namespace in order for the examples to be deployed onto CRC.
+
+## CLI
+### Installation
+You can build and install the Kogito CLI by running `make install-cli` 
+in the base directory of your cloned operator repository. 
+This will compile the CLI binary to `$GOPATH/bin` or `$HOME/go/bin` if the `GOPATH`
+environment variable is not set. To ensure that you can run 
+the binary, add `$GOPATH/bin`/`$HOME/go/bin` to your [`PATH` environment variable](https://opensource.com/article/17/6/set-path-linux).
+
+### Deployment
```

Comment:
I think we can make this more generic mentioning Minikube as well.

Reply:
Since this is in the CLI section, I opted to leave this part as is since the CLI only works with CRC. Instead, under the whole "[Deploying Examples to Kubernetes](https://github.com/Kevin-Mok/kogito-operator/blob/KOGITO-5127/docs/GUIDE_FOR_KOGITO_DEVS.md#deploying-examples-to-kubernetes)" section, I mentioned that the operator must be running in the current namespace and linked to the "[Running Operator Locally](https://github.com/Kevin-Mok/kogito-operator/blob/KOGITO-5127/docs/GUIDE_FOR_KOGITO_DEVS.md#Running-Operator-Locally)" section where it shows how to run the operator in CRC/minikube.